gds/src/main/scala/edu/gemini/aspen/gds/keywords/PropertyKeywordCollector.scala

```diff
@@ -0,0 +1,23 @@
+package edu.gemini.aspen.gds.keywords
+
+import cats.effect.Async
+import cats.syntax.all._
+import edu.gemini.aspen.gds.configuration.{ KeywordConfiguration, KeywordConfigurationItem }
+import edu.gemini.aspen.gds.fits.FitsValue
+import edu.gemini.aspen.gds.model.{ GdsError, KeywordSource }
+
+object PropertyKeywordCollector {
+  def apply[F[_]](config: KeywordConfiguration)(implicit F: Async[F]): KeywordCollector[F] = {
+    def retriever(item: KeywordConfigurationItem): F[FitsValue] =
+      Option(System.getProperty(item.channel.name)) match {
```

Comment:
OTOH maybe this call should be under `delay`

Reply:
Yeah. You're right, this should be under delay. `setProperty` can throw exceptions.